docs/api/ethereum.md

```diff
@@ -1,5 +1,645 @@
+
+**Arguments**
+
+| Argument | Description
+|-|-|-
+| options.assetLedgerSource | A `string` representing the URL to the compiled ERC-721 related smart contract definition file. This file is used when deploying new asset ledgers to the network.
+| options.clientId | A `string` representing the Bitski client id. You get the client ID by creating a [developer account](https://developer.bitski.com/) on Bitski.
+| options.mutationTimeout | A `number` representing the number of milliseconds in which a mutation times out. Defaults to `3600000`. You can set it to `-1` for disable timeout.
+| options.orderGatewayId | A `string` representing an Ethereum address of the [order gateway](/#public-addresses).
+| options.networkName | A `string` representing which ethereum network we will connect to.
+| options.redirectUrl | A `string` representing Bitski redirect URL. For Bitski front-end integration you will need to create a call back website for OAuth2 and host it. Here is an [example](https://github.com/BitskiCo/bitski-js/blob/develop/packages/browser/callback.html). The URL to this HTML page also needs to be apps authorized redirect URL in [Bitski Developer Portal](https://developer.bitski.com/). 
+| options.requiredConfirmations | An `integer` representing the number of confirmations needed for mutations to be considered confirmed. It defaults to `1`.
+| options.signMethod | An `integer` representing the signature type. The available options are `0` (eth_sign) or `2` (EIP-712) or `3` (perosnal_sign). It defaults to `0`.
+| options.unsafeRecipientIds | A list of `strings` representing smart contract addresses that do not support safe ERC-721 transfers (e.g. CryptoKitties address should be listed here). 
+| options.valueLedgerSource | A `string` representing the URL to the compiled ERC-20 related smart contract definition file. This file is used when deploying new value ledgers to the network.
+
+**Usage**
+
+```ts
+import { BitskiProvider } from '@0xcert/ethereum-bitski-frontend-provider';
+
+const provider = new BitskiProvider({
+    clientId: '',
+    redirectUrl: 'https://...',
+});
+```
+
+**See also:**
+
+[Bitski back-end provider](#bitski-back-end-provider)
+
+### accountId
+
+A class instance `variable` holding a `string` which represents user's current Ethereum wallet address.
+
+### assetLedgerSource
+
+A class instance `variable` holding a `string` which represents the URL to the compiled ERC-721 related smart contract definition file. This file is used when deploying new asset ledgers to the network.
+
+### emit(event, ...options);
+
+A `synchronous` class instance `function` to manually trigger a provider event.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| event | [required] A `string` representing a [provider event](./ethereum.md#provider-events) name.
+| options | [required] Pass valid current and previous account ID for `ACCOUNT_CHANGE` event or valid current and previous network version for `NETWORK_CHANGE` event.
```

Comment:
| options | [required] Pass valid current and previous account ID for `ACCOUNT_CHANGE` event or valid current and previous network versions for `NETWORK_CHANGE` event.